exo-go/src/app_runner/index.go

```diff
@@ -113,6 +114,19 @@ func (a *AppRunner) Shutdown(shutdownConfig types.ShutdownConfig) error {
 
 // Start runs the application and returns the process and returns an error if any
 func (a *AppRunner) Start() error {
+	watcherErr := make(chan error)
+	if err := a.watchServices(watcherErr); err != nil {
+		return err
+	}
+	go func() {
+		err := <-watcherErr
+		if err != nil {
+			if err := a.Shutdown(types.ShutdownConfig{CloseMessage: "Failed to restart"}); err != nil {
+				a.write("Failed to shutdown")
+			}
+		}
+	}()
```

Comment:
I think this goroutine and the watcherErr variable should live in the `watchServices` function.